src/terms/xc.jl

```diff
@@ -529,11 +529,19 @@ _matify(data::AbstractArray) = reshape(data, size(data, 1), :)
 
 for fun in (:potential_terms, :kernel_terms)
     @eval begin
-        function DftFunctionals.$fun(xc::Functional, density::LibxcDensities)
+        function DftFunctionals.$fun(xc::DispatchFunctional, density::LibxcDensities)
             $fun(xc, _matify(density.ρ_real), _matify(density.σ_real),
                      _matify(density.τ_real), _matify(density.Δρ_real))
         end
 
+        # Ensure functionals from DftFunctionals are sent to the CPU, until DftFunctionals.jl is refactored
```

Comment:
As discussed, change the comment to add TODO and mention that it can be removed once DftFunctionals is fixed.